src/plugins/plugins_ex.h

```diff
@@ -452,6 +452,10 @@ class pluginex : public plugin
     [[nodiscard]]
     std::unique_ptr<libhook::SyscallHook> createSyscallHook(const std::string& syscall_name, Callback cb);
 
+    template<typename Params = PluginResult, typename Callback>
+    [[nodiscard]]
+    std::unique_ptr<libhook::SyscallHook> createSyscallHook(const std::string& syscall_name, Callback cb, const std::string& display_name);
```

Comment:
Please don't. There are other type of hooks which all have unified api. Adding random `display_name` is not really wanted. You can either manually edit `trap->name` after creation, or store data in custom parameters via `PluginResult`.

Reply:
I understand your doubt, but it was done this way, because under linux traps are installed a little differently than under windows. Look at the `SyscallHook::create` method, it has slightly changed the logic for setting the trap, for example, `drakvuf_get_kernel_symbol_va` is used to get the symbol, `hook->trap_->breakpoint.module` specifies `linux`, instead of 'ntoskrnl.exe `. If you have any ideas on how this can be unified under one method, I am ready to listen to you.

Reply:
Well, it's kind of obvious, that `SyscallHook::create` should be fixed, as it has hardcoded `ntoskrnl.exe` and `pid = 4`. `drakvuf_get_os_type` should be called and filled with proper data depending on the OS.

As for setting `display_name` on the trap, it should be done outside of `libhook`. Create hook, get trap from inside, set field manually in plugin. Noone forbids you from doing that :)

Reply:
@BonusPlay @tklengyel  I can do it in separate MR. I propose to merge the current pull request as is.

Reply:
@disaykin I support this idea

Reply:
See MR#1516